drivers/scale_by_host_template_id.go

```diff
@@ -0,0 +1,404 @@
+	}
+
+	if config.Action == "up" {
+		if config.DeleteOption != "" {
+			return http.StatusBadRequest, fmt.Errorf("Delete option not to be provided while scaling up")
+		}
+	}
+
+	if config.Action == "down" {
+		if config.DeleteOption != "mostRecent" && config.DeleteOption != "leastRecent" {
+			return http.StatusBadRequest, fmt.Errorf("Invalid delete option/Delete option missing %v", config.DeleteOption)
+		}
+	}
+
+	return http.StatusOK, nil
+}
+
+func (s *ScaleByHostTemplateIDDriver) Execute(conf interface{}, apiClient *client.RancherClient, reqBody interface{}) (int, error) {
+	var currNameSuffix, baseHostName, currCloneName, suffix string
+	var count, index, newHostScale, baseHostIndex int64
+	httpClient := &http.Client{
+		Timeout: time.Second * 10,
+	}
+
+	config := &model.ScaleByHostTemplateID{}
+	err := mapstructure.Decode(conf, config)
+	if err != nil {
+		return http.StatusInternalServerError, errors.Wrap(err, "Couldn't unmarshal config")
+	}
+
+	action := config.Action
+	amount := config.Amount
+	min := config.Min
+	max := config.Max
+	deleteOption := config.DeleteOption
+
+	cattleConfig := rConfig.GetConfig()
+	cattleURL := cattleConfig.CattleURL
+	u, err := url.Parse(cattleURL)
+	if err != nil {
+		panic(err)
+	}
+	cattleURL = strings.Split(cattleURL, u.Path)[0] + "/v2-beta"
+
+	filters := make(map[string]interface{})
+	filters["sort"] = "created"
+	filters["order"] = "desc"
+	hostCollection, err := apiClient.Host.List(&client.ListOpts{
+		Filters: filters,
+	})
+	if len(hostCollection.Data) == 0 {
+		return http.StatusBadRequest, fmt.Errorf("No hosts for scaling found")
+	}
+
+	hostScalingGroup := []client.Host{}
+	baseHostIndex = -1
+	for _, host := range hostCollection.Data {
+		if config.HostTemplateID != "" {
+			// add hosts with specified hostTemplateId
+			if host.HostTemplateId == config.HostTemplateID {
+				hostScalingGroup = append(hostScalingGroup, host)
+				if host.Driver != "" {
+					baseHostIndex = int64(len(hostScalingGroup)) - 1
+				}
+			}
+		}
+	}
+
+	if baseHostIndex == -1 && action == "up" {
+		return http.StatusBadRequest, fmt.Errorf("Cannot use custom hosts for scaling up")
+	}
+
+	count = 0
+
+	if action == "up" {
+		// Consider the least recently created as base host for cloning
+		// Remove domain from host name, scaleHost12.foo.com becomes scaleHost12
+		// Remove largest number suffix from end, scaleHost12 becomes scaleHost
+		// Name has precedence over hostname. If name is set, empty this field for the clones
+		host := hostScalingGroup[baseHostIndex]
+		if host.Name != "" {
+			baseHostName = host.Name
+		} else {
+			baseHostName = host.Hostname
+		}
+		baseHostName = strings.Split(baseHostName, ".")[0]
+		baseSuffix := reg.FindString(baseHostName)
+		basePrefix := strings.TrimRight(baseHostName, baseSuffix)
+
+		// Use raw call to get host so as to get additional driver config
+		getURL := cattleURL + "/projects/" + host.AccountId + "/hosts/" + host.Id
```

Comment:
You no longer need to use raw http calls. We only did that because of a quirk with how the host api worked when you had to set the machine config (like digitaloceanConfig, amazonec2Config), but you don't have to set the config, you just need to set the hostTempplateId.

So, you just use the go-rancher client, like `client.Hosts.Create()`